test/templates/realtime.tpl.js

```diff
@@ -9,16 +9,35 @@ const kuzzle = new Kuzzle('websocket', {
 
 // add a listener to detect any connection problems
 kuzzle.on('networkError', error => {
-  console.error(`Network Error: ${error}`);
+  console.error(`Network Error: ${error.message}`);
 });
 
+function sleep (ms) {
+  return new Promise(resolve => setTimeout(resolve, ms));
+}
+
 (async () => {
   try {
     await kuzzle.connect();
   } catch (error) {
     console.log(`Cannot connect to Kuzzle: ${error.message}`);
   }
+
+  const consoleLog = console.log;
+  const outputs = [];
+
+  console.log = (...args) => {
+    outputs.push(...args);
+  };
+
   [snippet-code] finally {
-    setTimeout(() => kuzzle.disconnect(), 100);
+    while (outputs.length <= 0) {
```

Comment:
You need to break from the loop after a maximum number of retries, to prevent infinite loops if the test fails